### PR TITLE
feat(storage): add hostname cases to SignedURL tests

### PR DESCRIPTION
**Pull request by BrennaEpp**
- If a library does not have a certain option (eg. emulator host), in the tests it may set something that takes the next precedence or vice-versa, or skip tests that set it. From least to most precedence: emulator host -> endpoint on client -> hostname in options

### CODE REVIEW

**Review by tritone — APPROVED** [review]
LGTM. You should work with other language owners to get this propagated into their repos with the necessary changes to the test code.

**Review by danielbankhead — APPROVED** [review]
LGTM

File: storage/v1/v4_signatures.json [hunk 1]
```diff
@@ -377,6 +377,18 @@
...
+    },
+    {
+      "description": "Universe domain",
+      "bucket": "test-bucket",
+      "object": "test-object",
+      "method": "GET",
+      "expiration": 10,
+      "timestamp": "2019-02-01T09:00:00Z",
+      "expectedUrl": "https://storage.domain.com:443/test-bucket/test-object?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=test-iam-credentials%40dummy-project-id.iam.gserviceaccount.com%2F20190201%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20190201T090000Z&X-Goog-Expires=10&X-Goog-SignedHeaders=host&X-Goog-Signature=8cd0d479a88fb7d791a2dcc8fc5b5f020ca817eeef5b5a5cb3260eb63cf47ecd271faa238d0fa31efca35bc2a9244bd122178c520749f922c0235726a5a6be099bf4f33a0d54187eee2e0208964c2a13104b03e235cdeb4f07b3eb566b8a33259cf7540a3fe823be601ace2a54a79acd6834cb646380c4cfc7ef0fd95d3ebbc1f97d840f6fe1dceed4269ecb4e91ff7e6633f38adab82049a965968367b9e7c362cec868d804bd42abbb6d2e837ce5d45ee9e1d92c7acc09623acaae3df6128ca15f9f80bb6543944e8c997f691c35113b9e9f44e86fd343524343b08dd8f887685588acc103e0b432f24912e7e1c63e086aeed1890e41b75beb64164fe6bfcf",
+      "universeDomain": "domain.com:443",
```
Comment (amanda-tarafa) [inline storage/v1/v4_signatures.json:389]: Should the port be included as part of the universe domain?
That's not what libraries in general are expecting.

Comment (BrennaEpp) [reply storage/v1/v4_signatures.json]: Ah, good point! I became over-eager as ports are accepted in endpoints and we had a bug in the Go library related to that. Removed the port from the universe domain.

**Review by amanda-tarafa — APPROVED** [review]
LGTM.

If I run into something while implementing these, I'll let you know.

File: storage/v1/v4_signatures.json [hunk 2]
```diff
@@ -285,6 +285,123 @@
...
+      "scheme": "http",
+      "hostname": "localhost:8080",
+      "expectedCanonicalRequest": "GET\n/test-bucket/test-object\nX-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=test-iam-credentials%40dummy-project-id.iam.gserviceaccount.com%2F20190201%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20190201T090000Z&X-Goog-Expires=10&X-Goog-SignedHeaders=host\nhost:storage.googleapis.com\n\nhost\nUNSIGNED-PAYLOAD",
```
Comment (amanda-tarafa) [inline storage/v1/v4_signatures.json:312]: Shouldn't the bit `host:storage.googleapis.com` in `expectedCanonicalRequest` be `host:localhost:8080`?

File: storage/v1/v4_signatures.json [hunk 3]
```diff
@@ -285,6 +285,123 @@
...
+      "scheme": "http",
+      "clientEndpoint": "http://localhost:8080",
+      "expectedCanonicalRequest": "GET\n/test-bucket/test-object\nX-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=test-iam-credentials%40dummy-project-id.iam.gserviceaccount.com%2F20190201%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20190201T090000Z&X-Goog-Expires=10&X-Goog-SignedHeaders=host\nhost:storage.googleapis.com\n\nhost\nUNSIGNED-PAYLOAD",
```
Comment (amanda-tarafa) [inline storage/v1/v4_signatures.json:338]: And here as well?

File: storage/v1/v4_signatures.json [hunk 4]
```diff
@@ -285,6 +285,123 @@
...
+      "emulatorHostname": "https://xyz.googleapis.com",
+      "expectedCanonicalRequest": "GET\n/test-bucket/test-object\nX-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=test-iam-credentials%40dummy-project-id.iam.gserviceaccount.com%2F20190201%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20190201T090000Z&X-Goog-Expires=10&X-Goog-SignedHeaders=host\nhost:storage.googleapis.com\n\nhost\nUNSIGNED-PAYLOAD",
```
Comment (amanda-tarafa) [inline storage/v1/v4_signatures.json:350]: And here it should be `xyz.googleapis.com`?

**Review by amanda-tarafa — COMMENTED** [review]
@BrennaEpp I've left a couple of questions here, thanks!

File: storage/v1/v4_signatures.json (hunk 2 above)
Comment (BrennaEpp) [reply storage/v1/v4_signatures.json:312]: You are correct (here and elsewhere); I just found out that we don't test this field in Go (we'd have to go into the internals of a function to test). I will update and send a PR for review

**Pull request merged**